src/main/resources/liquibase/workbench_changelog/12_5_0.xml

```diff
@@ -0,0 +1,713 @@
+		<sql dbms="mysql" splitStatements="true">
+			INSERT INTO role_permission (permission_id, role_id) VALUES (1,1);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (1,5);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (10,2);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (4,2);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (5,2);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (6,2);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (10,3);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (4,3);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (5,3);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (6,3);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (14,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (17,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (18,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (19,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (20,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (21,4);
+			INSERT INTO role_permission (permission_id, role_id) VALUES (22,4);
+		</sql>
+	</changeSet>
+
+
+	<changeSet author="vanina" id="v12.5.0-22">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				select count(1) from role where id = 5;
+			</sqlCheck>
+		</preConditions>
+		<comment>
+			Update table role where row is SUPERADMIN
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE role SET editable = 0, assignable = 0 WHERE id = 5;
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v12.5.0-23">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<sqlCheck expectedResult="0">
+					select count(1) from users_roles
+					where role_id in (2, 3) and crop_name is null;
+				</sqlCheck>
+			</not>
+		</preConditions>
+		<comment>Expand breeder and technician roles rows to apply to all user crops, removing rows with empty crop_name</comment>
+		<sql dbms="mysql" splitStatements="true">
+			insert into users_roles (userid, role_id, crop_name)
+			select ur.userid, ur.role_id, uc.crop_name
+			from users_roles ur
+					 inner join users_crops uc on (uc.user_id = ur.userid)
+			where ur.role_id in (2, 3);
+			delete from users_roles
+			where role_id in (2, 3) and crop_name is null;
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v12.5.0-24">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<sqlCheck expectedResult="0">
+					select count(1) from users_roles
+					where role_id = 4 and crop_name is null
+				</sqlCheck>
+			</not>
+		</preConditions>
+		<comment>Associate the readonly role to the crop and programs which user is member of, given that readonly role is a program role</comment>
+		<sql dbms="mysql" splitStatements="true">
+			insert into users_roles (userid, role_id, crop_name, workbench_project_id)
```

Comment:
Same here, won't it be simpler to do an UPDATE statement instead of INSERT then DELETE? Plus we don't need to join to **workbench_crop** to get the crop name.
```
 UPDATE users_roles ur
			 inner join workbench_project_user_info wpui on (wpui.user_id = ur.userid)
			 inner join workbench_project wp on (wp.project_id = wpui.project_id)
			 SET ur.crop_name =  wp.crop_type,
                 workbench_project_id = wpui.project_id
			where ur.role_id = 4;
```